src/RealPath.php

```diff
@@ -24,74 +23,28 @@
 
 /**
  * @psalm-type Options = array{
- *     exists: bool,
- *     ...
+ *     exists?: bool,
  * }
- * @template TOptions of Options
- * @extends AbstractFilter<TOptions>
+ * @implements FilterInterface<string|array<array-key, string|mixed>>
  */
-final class RealPath extends AbstractFilter
+final class RealPath implements FilterInterface
 {
-    /** @var TOptions $options */
-    protected $options = [
-        'exists' => true,
-    ];
+    private readonly bool $pathMustExist;
 
-    /**
-     * @param  bool|Traversable|Options $existsOrOptions Options to set
-     */
-    public function __construct($existsOrOptions = true)
+    /** @param Options $options */
+    public function __construct(array $options = [])
     {
-        if ($existsOrOptions !== null) {
-            if (! static::isOptions($existsOrOptions)) {
-                $this->setExists($existsOrOptions);
-            } else {
-                $this->setOptions($existsOrOptions);
-            }
-        }
-    }
-
-    /**
-     * Sets if the path has to exist
-     * TRUE when the path must exist
-     * FALSE when not existing paths can be given
-     *
-     * @param  bool $flag Path must exist
-     * @return self
-     */
-    public function setExists($flag = true)
-    {
-        $this->options['exists'] = (bool) $flag;
-        return $this;
+        $this->pathMustExist = $options['exists'] ?? true;
     }
 
-    /**
-     * Returns true if the filtered path must exist
-     *
-     * @return bool
-     */
-    public function getExists()
-    {
-        return $this->options['exists'];
-    }
-
-    /**
-     * Defined by Laminas\Filter\FilterInterface
-     *
-     * Returns realpath($value)
-     *
-     * If the value provided is non-scalar, the value will remain unfiltered
-     *
-     * @psalm-return ($value is string ? string : mixed)
-     */
     public function filter(mixed $value): mixed
     {
         if (! is_string($value)) {
             return $value;
         }
         $path = (string) $value;
 
-        if ($this->options['exists']) {
+        if ($this->pathMustExist) {
             return realpath($path);
         }
 
```

Comment:
Can we get rid of `ErrorHandler::start()` and `stop()` here? I'm not convinced that any warnings/errors are emitted: https://3v4l.org/jT1XV

Reply:
Also for all the code below, it remains untested because CI does not run windows 🤔

Reply:
> Can we get rid of `ErrorHandler::start()` and `stop()` here? I'm not convinced that any warnings/errors are emitted: https://3v4l.org/jT1XV

Looks like we can. I played around with the open_baseline restriction just to make sure it wasn't something to do with that, but it also just returns false if outside of the allowed directories.